test/server/graphql/v2/query/TransactionsQuery.test.js

```diff
@@ -62,4 +96,61 @@ describe('server/graphql/v2/query/TransactionsQuery', () => {
     result = await graphqlQueryV2(transactionsQuery, { ...queryParams, maxAmount: 20 });
     expect(result.data.transactions.totalCount).to.eq(1);
   });
+
+  describe('permissions', () => {
+    it('can seel legalName if owner or host admin', async () => {
+      const randomUser = await fakeUser();
+      const queryArgs = { slug: collective.slug };
+
+      const resultUnauthenticated = await graphqlQueryV2(transactionsQuery, queryArgs);
+      resultUnauthenticated.data.transactions.nodes.forEach(transaction => {
+        expect(transaction.fromAccount.legalName).to.be.null;
+        expect(transaction.oppositeAccount.legalName).to.be.null;
+        expect(transaction.account.legalName).to.be.null;
+        expect(transaction.toAccount.legalName).to.be.null;
+      });
+
+      const resultRandomUser = await graphqlQueryV2(transactionsQuery, queryArgs, randomUser);
+      resultRandomUser.data.transactions.nodes.forEach(transaction => {
+        expect(transaction.fromAccount.legalName).to.be.null;
+        expect(transaction.oppositeAccount.legalName).to.be.null;
+        expect(transaction.account.legalName).to.be.null;
+        expect(transaction.toAccount.legalName).to.be.null;
+      });
+
+      const resultCollectiveAdmin = await graphqlQueryV2(transactionsQuery, queryArgs, collectiveAdmin);
+      resultCollectiveAdmin.data.transactions.nodes.forEach(transaction => {
+        expect(transaction.fromAccount.legalName).to.be.null;
+        expect(transaction.oppositeAccount.legalName).to.be.null;
+        expect(transaction.account.legalName).to.be.null;
+        expect(transaction.toAccount.legalName).to.be.null;
+      });
+
+      const resultFromCollectiveAdmin = await graphqlQueryV2(transactionsQuery, queryArgs, fromCollectiveAdmin);
+      resultFromCollectiveAdmin.data.transactions.nodes.forEach(transaction => {
+        if (transaction.type === 'CREDIT') {
```

Comment:
That doesn't matter much in tests. Especially since this one has, and will ever have, only two values: `CREDIT` and `DEBIT`. There's no reason for us to change them in the future. Moreover, typing this in typescript should ideally be:

```ts
type TransactionType = 'CREDIT' | 'DEBIT'
```

No enum involved. We can eventually start an RFC if you think we should enforce the use of enums everywhere, but as such I don't have the feeling that it brings any value here.

Reply:
Makes sense. I try to use enums everywhere but I think this is a minor detail. 👍🏽